annotator/tests/test_frame_extraction.py

```
@@ -0,0 +1,271 @@
+
+    monkeypatch.setattr(base.subprocess, "run", run_then_let_others_evict)
+
+    target = tmp_path / "target.mp4"
+    target.write_bytes(b"x" * 32)
+
+    result = {}
+    def reader():
+        frames = list(base.iter_frames(target, fps=0.5))
+        result["ok"] = len(frames) == 2 and all(p.exists() for _, p in frames)
+
+    t = _t.Thread(target=reader); t.start()
```

Ruff (0.16.0)

[error] Multiple statements on one line (semicolon)

(E702)